src/App.js

```diff
@@ -851,6 +851,24 @@ export class App extends React.Component<AppProps, AppState> {
         this.focusTrapManager.handleKeyDown(e);
     }
 
+    handleRootFocus = (e: SyntheticKeyboardEvent<HTMLElement>) => {
+        // $FlowFixMe getAttribute is missing in event target
+        const ariaLabel = e.target.getAttribute('aria-label');
+        if (!this.audioManager.getFeedbackIsPlaying()) {
+            if (ariaLabel) {
+                this.audioManager.playPreviewAnnouncement(ariaLabel);
+            } else {
+                // $FlowFixMe innerText is missing in event target
+                const innerText = e.target.innerText;
+                if (innerText) {
+                    this.audioManager.playPreviewAnnouncement(innerText);
+                }
+            }
+        } else {
+            this.audioManager.setFeedbackIsPlaying(false);
```

Comment:
This seems incorrect.  If a feedback announcement is playing, the first pass through this will clear the flag, so depending on the length of the feedback announcement, you might still attempt to play a preview announcement anyway.

I'd suggest setting the flag to false using the [`end` event](https://developer.mozilla.org/en-US/docs/Web/API/SpeechSynthesisUtterance/end_event) for an utterance instead.  One way to handle this would be to support an optional callback (to clear the event) as a second argument to `playStringMessage`.

Reply:
That is exactly what I want; clear the flag during the first pass through. I think feedback and preview announcement have equal importance, so any of the two announcements can cancel each other.

Reply:
OK, thank you for explaining further, maybe a comment above the check explaining the rationale is enough?  i.e. without the flag/check we do X too often, et cetera.